pybossa/view/projects.py

```diff
@@ -1343,6 +1344,19 @@ def respond(tmpl):
         if has_no_presenter(project):
             flash(gettext("Sorry, but this project is still a draft and does "
                           "not have a task presenter."), "error")
+
+        # Get locked task for this project.
+        timeout = project.timeout
+        task_id, remaining_time = get_task_id_and_duration_for_project_user(project.id, user_id)
+        if task_id and remaining_time:
```

Comment:
1. We need to make the logic consistent with the scheduler code: https://github.com/bloomberg/pybossa/blob/b5d7cbb06b0e9650b03211b53ba2509837b06afc/pybossa/sched.py#L273-L276

Therefore, the condition could be:
```
if task_id and remaining_time > 10:
    # Set the original timeout seconds to display in the message.
    ....
```

2. In case the IF statement is not executed, we need to set default value for `template_args['project'].original_timeout` 
so the code might be:
```
template_args['project'].original_timeout = timeout
if task_id and remaining_time > 10:
    # Set the original timeout seconds to display in the message.
    ....
```

Reply:
https://github.com/bloomberg/pybossa/pull/734/commits/8570e7357e74b2c558f06b7edb97935e84ecea23

Reply:
>  could you elaborate on the purpose of checking if the remaining time is > 10 seconds? Thanks!

This aligns with our scheduler logic. When the user requests a task from "start contributing now", GIGwork would do the following 

If there are tasks locked by the user:
    fetch the task with the longest remaining time
    If the longest remaining time is more than 10 seconds, give this task to the user.
    If the longest remaining time is no more than 10 seconds, query db and find a new task for the user.
if there are no tasks locked by the user:
    query db and find a new task for user.

I think we add the 10-second condition to avoid bad user experience - we are assuming that most of users need more than 10 seconds to answer a task, therefore, if the task remaining time is less than 10 second, we rather give a new task with full timeout window.